packages/accounts-password/password_server.js

```diff
@@ -1146,25 +1146,42 @@ Meteor.methods({createUser: function (...args) {
           error: new Meteor.Error(403, "Signups forbidden")
         };
 
-      // Create user. result contains id and token.
-      const userId = createUser(options);
-      // safety belt. createUser is supposed to throw on error. send 500 error
-      // instead of sending a verification email with empty userid.
-      if (! userId)
-        throw new Error("createUser failed to insert new user");
-
-      // If `Accounts._options.sendVerificationEmail` is set, register
-      // a token to verify the user's primary email, and send it to
-      // that address.
-      if (options.email && Accounts._options.sendVerificationEmail)
-        Accounts.sendVerificationEmail(userId, options.email);
+      const userId = Accounts.createUserVerifyingEmail(options);
 
       // client gets logged in as the new user afterwards.
       return {userId: userId};
     }
   );
 }});
 
+// Create user directly on the server.
+//
+// Differently from Accounts.createUser(), this evaluates the Accounts package
+// configurations and send a verification email if the user has been registered
+// successfully.
+Accounts.createUserVerifyingEmail = (options) => {
+  options = { ...options };
```

Comment:
Why do we need this line?

Reply:
_createUser_ [original implementation](https://github.com/meteor/meteor/blob/6dfa1a6a9f63e7259e0f2383075238207be3dc52/packages/accounts-password/password_server.js#L1190) does this. I think it is related to immutability but idk for sure.